client/src/featureform/get_local.py

```diff
@@ -0,0 +1,400 @@
+    ("STATUS: ", returned_training_set["status"])])
+    format_new_para("LABEL: ")
+    format_rows([("NAME", "VARIANT"), (returned_training_set["label"]["name"], returned_training_set["label"]["variant"])])
+    format_new_para("FEATURES:")
+    format_rows("NAME", "VARIANT")
+    for f in returned_training_set["features"]:
+        format_rows(f["name"], f["variant"])
+    format_new_para("")
+    return returned_training_set
+
+def get_provider_info_local(name):
+    db = SQLiteMetadata()
+    providers_list = db.getVariantResource("providers", "name", name)
+    if providers_list == []:
+        print("Provider not found.")
+        return
+    provider = providers_list[0]
+
+    returned_features_list = get_related_resources("feature_variant", 
+                                                    name_col="name",
+                                                    var_col="variant",
+                                                    column1="provider",
+                                                    name=name)
+                                                    
+    returned_labels_list = get_related_resources("label_variant", 
+                                                    name_col="name",
+                                                    var_col="variant",
+                                                    column1="provider",
+                                                    name=name)
+    returned_provider = {
+        "name": provider["name"],
+        "type": provider["type"],
+        "description": provider["description"],
+        "type": provider["provider_type"],
+        "software": provider["software"],
+        "team": provider["team"],
+        "status": provider["status"],
+        "serializedConfig": provider["serialized_config"],
+        "sources": provider["sources"],
+        "features": returned_features_list,
+        "labels": returned_labels_list
+    }
+    format_rows([("NAME: ", returned_provider["name"]),
+    ("DESCRIPTION: ", returned_provider["description"]),
+    ("TYPE: ", returned_provider["type"]),
+    ("SOFTWARE: ", returned_provider["software"]),
+    ("TEAM: ", returned_provider["team"]),
+    ("STATUS: ", returned_provider["status"])])
+    format_rows("SOURCE:", returned_provider["sources"])
+    format_new_para("FEATURES:")
+    format_rows("NAME", "VARIANT")
+    for f in returned_provider["features"]:
+        format_rows(f["name"], f["variant"])
+    format_new_para("LABELS:")
+    format_rows("NAME", "VARIANT")
+    for l in returned_provider["labels"]:
+        format_rows(l["name"], l["variant"])
+    format_new_para("")
+    return returned_provider
+
+def get_related_resources(table, column1, name, column2=None, variant=None, name_col="name", var_col="variant", format=True):
+    db = SQLiteMetadata()
+    if column2 and variant:
+        try:
+            res_list = db.getNameVariant(table, column1, name, column2, variant)
```

Comment:
@ahmadnazeri is there a cleaner way of doing this?

Reply:
@sdreyer and @steffitan23, I would move the `try...except` outside of the if conditions. 

```
try: 
     if column2 and variant:
                ...
     else:
                ...
except ValueError:
        res_list = []

if format:
      res_list = [...]

return res_list
```

What do you think?